src/widget-core/Registry.ts

```diff
@@ -87,16 +95,23 @@ export interface RegistryInterface {
  * @param item the item to check
  * @returns true/false indicating if the item is a WidgetBaseConstructor
  */
-export function isWidgetBaseConstructor<T extends WidgetBaseInterface>(item: any): item is Constructor<T> {
+export function isWidgetBaseConstructor<T extends WidgetBaseInterface = any>(item: any): item is Constructor<T> {
 	return Boolean(item && item._type === WIDGET_BASE_TYPE);
 }
 
+export function isWidgetFunction(item: any): item is Callback<any, any, RenderResult> {
+	return Boolean(item && item.isWidget);
+}
+
+export function isWidget<T extends WidgetBaseInterface = any>(
+	item: any
+): item is Constructor<T> | Callback<any, any, RenderResult> {
+	return isWidgetBaseConstructor(item) || isWidgetFunction(item);
+}
+
 export function isWidgetConstructorDefaultExport<T>(item: any): item is ESMDefaultWidgetBase<T> {
 	return Boolean(
-		item &&
-			item.hasOwnProperty('__esModule') &&
-			item.hasOwnProperty('default') &&
-			isWidgetBaseConstructor(item.default)
+		item && item.hasOwnProperty('__esModule') && item.hasOwnProperty('default') && isWidget(item.default)
```

Comment:
I think for code splitting this needs to also check whether it's a widget factory.

Reply:
Updated